### ISSUE TITLE
Refactoring draw-map into the app component structure

### ISSUE THREAD

**Issue by Don-Isdale** [plantinformatics/pretzel#190]
##### Description
- integrate the stacks & axes more closely with the Ember components which are now the driving force of the app.

##### Background
Stacks & axes were written as javascript classes, using imperative update;  as the number of components and interactions grew, 
a shift was made to implement new components using functional reactive logic (Computed Properties), which is not as efficient as imperative logic, but it scales better with many complex interactions - as long as the connections and dependencies are declared correctly, it will automatically update / recalculate when required, using lazy evaluation.
For the task of moving functions out of draw-map into separate components : most of these functions are related to stacks & axes, so it is best to start by integrating stacks & axes functions into the existing components (stacks-view and axis-1d), so that they are acting merely as utility / library functions, rather than providing imperative actions driving the app event flow.
This change is going well, the core part is done.

##### Methodologies
Moving a function requires re-establishing the input and output data connections, and
providing the facilities which the function being moved depends on.

To some extent it is useful to provide a facsimile of those dependencies / affordances, based on the new design; where there are many uses of a particular dependency, this can save time.
That approach can be transitional;  if changing the interfaces improves the design, that can be done progressively at lower risk.

This method is used when it is important to minimise the implementation effort and the risk of changes requiring secondary changes,
which is not really the case in this instance;  may use this technique for some transitional scaffolding, with the benefit that having the GUI functional is useful for testing.
(for interest, this method was labelled Strangler Fig by Martin Fowler, after he visited Qld rainforests).
(detail of this scaffolding has been prepended to #190 first [comment](https://github.com/plantinformatics/pretzel/issues/190#issue-674013386) )

##### Re-use / Re-write decisions
Each of the functions was evaluated in terms of its value - whether it was exactly or close to the final requirement;  this enables a decision about the degree of reuse / rewrite.
This task #190 lists the functions which are to be moved (i.e. retained), and at the bottom of the list there is a section labelled 'not required' - these functions provide features which are equivalently provided by newer and preferred facilities.
Some other functions have notes indicating how they can be simplified;  the remainder require only the changes for inputs / output and supporting interfaces, which as discussed can be relatively "mechanical".

The functions are listed roughly in parallel to the order in which they appear in the source code, for more efficient work and to check completeness.
As the work is done they can be listed in checked-off order, for easier reference by management to the progress.

##### Risks
- the amount of changes in the functions being moved may be greater than expected
  To quantify this, we can analyse in more detail the supporting dependencies required by functions being moved, list these and a design for supporting them in their new position.
  This has already done with a quick read through of the functions, noting their requirements and thinking about how they would be supported - there were none of any signicant difficulty.

- integration of stacks classes into ember components, and replacing some imperative actions by functional reactive evaluation, changes (improves) the flow of actions / data in the app - this could introduce a problem.
The expectation is that there won't be a problem; it is likely to be easier to make the app more responsive, and reduce duplication of processing which can occur in imperative flows because data is prepared without knowing when it will be used.

- In general terms, this is a significant architectural change to a moderately complex app, and so it is reasonable to expect impacts on functionality;  problems arising are likely to find reasonable solutions; i.e. it is not expected that they will be show-stoppers.

The expected time of each task follows approximately a Bell curve, with the mean value quoted in the estimate.
Considering the above risks, the tasks are 75% likely to complete within 2x that time, and 95% likely to complete in 4x that time.

###### Fallback plan
If a high priority feature / change is requested, or if time blows out, stash the branch and continue on to other work without merging any changes;
the work is still intact, and can be merged as time permits - it is desirable to minimise the delay in doing this because the merge gets harder as development continues on the feature branches.

#### Planning
There are various approaches to scheduling software maintenance :
- if it is embedded in feature work, it may seem to management that apparently simple features become black holes consuming time
- some tasks can't be done in pieces, and putting it on the schedule as a block of work makes it difficult to get time allocated, as even if it is seen as important, feature development work is more urgent.

---

#### Overview of progress / plan

(Not including functions already completed at 2022Nov10) 

| Status | Time | Task / Function Name |
|-|-|-|
| <ul><li>[x] </li></ul> | [/24H] |  <br />  da8214dd : Add stack-view, move axes functions from stack-view.js to axes-1d.js.   <br />  7e81aa42 : draft of continuing changes feature/axisFinishSplit3 |
| <ul><li>[x] </li></ul> | [16-20H] | integrate stacks.js functions with components, so that Block, Stacked, Stack will be synonymous with block-view, axis-1d, stack-view respectively <br />    i.e. the stacks.js functions provide views of the (Ember) data objects <br /> feature/axisFinishSplit3 <br />  8efd3dd6 : use draw/stack-view for portions(), positions(). <br />  eb1364e4 : connected stack-view and axis-1d. |
| <ul><li>[x] </li></ul> | [8H/8H] | move position calculation into models/draw/stack <br /> use draw/stack-view for portions(), positions(). 8efd3dd6 |
| <ul><li>[x] </li></ul> |  | **move to axis-2d :** |
| <ul><li>[x] </li></ul> | [/1H] | axisWidthResize() |
| <ul><li>[x] </li></ul> | [/1H] | axisWidthResizeEnded() |
| <ul><li>[x] </li></ul> | [/1H] | getAxisExtendedWidth() |
| <ul><li>[x] </li></ul> | [/1H] | axisShowExtend() |
| <ul><li>[x] </li></ul> | [2-3D/4H] | assign stacks.js functions to components : stack-view (added), and stacks-view |
| <ul><li>[x] </li></ul> | [/24H] | Move axis rendering to stack-view / axis-1d :  <br />  82d9151f : use draw/stack object without wrapper components/draw/stack-view  <br />  ccba73d6 : rename draw/stack.js from models/ to utils/  <br />  4315f635 : split FeatureTicks out of axis-1d.js  <br />  cbcc62dd : change path models to utils for stack |
| <ul><li>[x] </li></ul> | [2H] | move axis rendering to utils/draw/axis-draw.js   <br />  4847b4d6 : move stack / axis rendering to axis-draw |
| <ul><li>[x] </li></ul> | [2H] | axisChangeGroupElt() |
| <ul><li>[x] </li></ul> | [/3H] | **move to added components/draw/stacks-drag (possibly part of stacks-view) :** 5a8f9373 |
| <ul><li>[x] </li></ul> | [2H] | dragstarted() |
| <ul><li>[x] </li></ul> | [2H] | dragged() |
| <ul><li>[x] </li></ul> | [1H] | draggedAxisRedraw()  (parts may go to path-data), |
| <ul><li>[x] </li></ul> | [/8H] | **move to added components/draw/axis-brush-zoom :** 2dc8d7b8, f105e308, 4056e0b2 |
| <ul><li>[x] </li></ul> | [1H] | getBrushExtents() |
| <ul><li>[x] </li></ul> | [1H] | getBrushedRegions() |
| <ul><li>[x] </li></ul> | [1H] | axisBrushedDomain() |
| <ul><li>[x] </li></ul> | [1H] | axisRange2DomainFn() |
| <ul><li>[x] </li></ul> | [1H] | axisRange2Domain() |
| <ul><li>[x] </li></ul> | [1H] | axisBrushShowSelection() |
| <ul><li>[x] </li></ul> | [2H] | brushHelper() |
| <ul><li>[x] </li></ul> | [2H] | resetZooms | 
| <ul><li>[x] </li></ul> | [2H] | resetBrushes() |
| <ul><li>[x] </li></ul> | [1H] | removeBrushExtent() |
| <ul><li>[x] </li></ul> | [2H] | resetZoom() |
| <ul><li>[x] </li></ul> | [2H] | handleFeatureCircleMouseOver() |
| <ul><li>[x] </li></ul> | [1H] | handleFeatureCircleMouseOut() |
| <ul><li>[x] </li></ul> | [1H] | brushEnableFeatureHover() |
| <ul><li>[x] </li></ul> | [2H] | zoom() |
| <ul><li>[x] </li></ul> | [1H] | axisScaleChangedRaf() |
| <ul><li>[x] </li></ul> | [2H] | axisScaleChanged() |
| <ul><li>[x] </li></ul> | [1H] | brushended() |
| <ul><li>[x] </li></ul> | [2H] | draw_flipRegion |
| <ul><li>[x] </li></ul> | [2H] | features2Limits() |
| <ul><li>[x] </li></ul> | [2H] | flipRegionInLimits() |
| <ul><li>[x] </li></ul> | [1H] | triggerZoomedAxis() |
| <ul><li>[x] </li></ul> | [1H] | throttledZoomedAxis() |
| <ul><li>[x] </li></ul> | | **move to added components/draw/block-view-1d, (or possibly utils/draw/zoomPanCalcs.js)  :** 64447e32 |
| <ul><li>[x] </li></ul> | [1H/0.25H] | wheelDelta() |
| <ul><li>[x] </li></ul> | [2H/0.25H] | zoomFilter(d) |
| <ul><li>[x] </li></ul> | [2H/1H] | move DropTarget() to drop-target.js |
| <ul><li>[x] </li></ul> | [/4H] | **move to axis-1d :** 3aa14753, d21f6faa, bf4ceded |
| <ul><li>[x] </li></ul> | [1H] | axisTitle() |
| <ul><li>[x] </li></ul> | [2H] | axisTitleFamily() |
| <ul><li>[x] </li></ul> | [1H] | updateAxisTitles() |
| <ul><li>[x] </li></ul> | [1H] | updateAxisTitleSize() (if needed)  |
| <ul><li>[x] </li></ul> | [/4H] | **move to (added) axis-menu-actions.js :** 3aa14753 <br /> axisDelete(), axisFlip(), axisPerpendicular(),  axisExtend(), <br />  (block actions) blockUnview(), blockVisible(). |
| <ul><li>[x] </li></ul> | [/3H] | **move to added components/draw/block-selected-features : ** 4c0af3c7 |
| <ul><li> </li></ul> | [2H] | updatedSelectedFeatures() |
| <ul><li> </li></ul> | [1H] | sendUpdatedSelectedFeatures() |
| <ul><li>[x] </li></ul> | [1H/1H] | selectedFeatures_clear() |
| <ul><li>[x] </li></ul> | [1H/1H] | selectedFeatures_removeAxis() 38845cf4 |
| <ul><li>[x] </li></ul> | [/3H] | **move to added utils/draw/draw-controls :** (these can be reduced by using <input type="checkbox">, replacing need for setupToggle(). move drawOptions to view-controls, using oninput=action in place of setupToggle*(). 50042ca3, d80d30df. removed these : |
| <ul><li>[x] </li></ul> | [1H/0H] | setupTogglePathUpdate() |
| <ul><li>[x] </li></ul> | [1H/0H] | setupToggleModePublish() |
| <ul><li>[x] </li></ul> | [1H/0H] | setupToggleShowPathHover() |
| <ul><li>[x] </li></ul> | [1H/0H] | setupToggleShowAll() |
| <ul><li>[x] </li></ul> | [1H/0H] | setupToggleShowSelectedFeatures() |
| <ul><li>[x] </li></ul> | [1H/0H] | setupInputRange() |
| <ul><li>[x] </li></ul> | [1H/0H] | setupPathOpacity() |
| <ul><li>[x] </li></ul> | [1H/0H] | setupPathWidth() |
| <ul><li>[x] </li></ul> | [1H/0H] | updateSbSizeThresh() |
| <ul><li>[x] </li></ul> | [1H/0H] | setupSbSizeThresh() |
| <ul><li>[x] </li></ul> | [1H/0H] | setupVariousControls() |
| <ul><li>[x] </li></ul> |  | **move to added utils/draw/axisTitle (not axis-menu) :** e3c3d5a3 |
| <ul><li>[x] </li></ul> | [1H] | configureAxisTitleMenu() |
| <ul><li>[x] </li></ul> | [1H/0H] | removed configureAxisSubTitleMenu() |
| <ul><li>[x] </li></ul> |  | **move to utils/utility-chromosome.js : **  f4da4f44 | 
| <ul><li>[x] </li></ul> | [1H/0.1H] | mapChrName2Axis() |
| <ul><li>[x] </li></ul> | [1H/0.1H] | axisName2Chr() |
| <ul><li>[x] </li></ul> | [1H/0.1H] | cmName[] |
| <ul><li>[x] </li></ul> | [2H/0.1H] | axisName2MapChr() |
| <ul><li>[x] </li></ul> | [1H/0.1H] | makeMapChrName() |
| <ul><li>[x] </li></ul> | [1H/0.1H] | makeIntervalName() |
| <ul><li>[x] </li></ul> | [1H] | showTickLocations() to feature-info.js |
| <ul><li>[x] </li></ul> | [2H/0.5H] | configureHorizTickHover() to hover.js as configureHorizTickHover_orig. |
| <ul><li>[x] </li></ul> |  | **move to draw/path-data and axis-1d :** f43d5eb2 |
| <ul><li>[x] </li></ul> | [1H/0.5H] | blockO() |
| <ul><li>[x] </li></ul> | [1H/0.5H] | featureLine2() |
| <ul><li>[x] </li></ul> | [1H/0.5H] | inside() |
| <ul><li>[x] </li></ul> | [1H/0.5H] | pointSegment() |
| <ul><li>[x] </li></ul> | [2H/0.5H] | featureLineS{,2,3}() |
| <ul><li>[x] </li></ul> | [1H/0.5H] | lineHoriz() |
| <ul><li>[x] </li></ul> | [2H/0.5H] | path{,U,Ug}() |
| <ul><li>[x] </li></ul> | [2H/0.5H] | pathAliasGroup() |
| <ul><li>[x] </li></ul> | [1H/0.5H] | featureAliasesText() |
| <ul><li>[x] </li></ul> | [1H/0.5H] | inRangeI() |
| <ul><li>[x] </li></ul> | [2H/0.5H] | pathFeatureStore() |
| <ul><li>[x] </li></ul> | [2H/0.5H] | patham{,2}() |
| <ul><li>[x] </li></ul> | [2H/0.5H] | featureY_() |
| <ul><li>[x] </li></ul> | [1H/0.5H] | log_path_data() |
| <ul><li>[x] </li></ul> | [2H/0.5H] | pathUpdate_() |
| <ul><li>[x] </li></ul> | [2H/0.5H] | pathUpdateFlow() |
| <ul><li>[x] </li></ul> | [2H/0.5H] | pathUpdate() |
| <ul><li>[x] </li></ul> | f43d5eb2 | **move to view-controls.js :**    allowPathsOutsideZoom |
| <ul><li>[x] </li></ul> | f43d5eb2 | **move to added utils/draw/path-classes.js :**  <br />  pathColourDefault, use_path_colour_scale, path_colour_scale_domain_set, path_colour_scale,   showScaffoldFeatures (showScaffoldMarkers). showAsymmetricAliases is in flow-controls viewOptions,   setup of path_colour_{domain,scale} factored to form configurePathColour() and colouredFeaturesChanged(), featureScaffold, scaffolds, scaffoldFeatures, intervals, intervalNames, intervalTree, scaffoldTicks,  pathClassA(),    colouredAg(), classFromSet(), locationClasses(), pathClasses(), pathColourUpdate(), scaffoldLegendColourUpdate(),   clearScaffoldColours(),   getUsePatchColour() as getUsePathColour(). |
| <ul><li>[x] </li></ul> | f43d5eb2 | **move to added utils/draw/path-data.js :**  <br />  pathDataIsLine, pathDataInG,  showHoverExtraText, tracedAxisScale,  featureNameOfData(), data_text(),   featureNameInRange(), featureInRange(),    axisFeatureTick(),    dataOfPath(), featureNameOfPath(). |
| <ul><li>[x] </li></ul> | f43d5eb2 | **move to added utils/draw/path-info.js :**  <br />  toolTip, toolTipHovered, setupToolTip() factored from : Setup the path hover tool tip, setupMouseHover(), toolTipMouseOver(), toolTipMouseOut(), closeToolTip(), setupToolTipMouseHover(), handleMouseOver(), hidePathHoverToolTip(), handleMouseOut(), |
| <ul><li>[x] </li></ul> | f43d5eb2 | **move to added utils/draw/feature-info.js :**  <br />  showTickLocations(), splitting out showTickLocationsOfAxis(). |
| <ul><li>[x] </li></ul> | [3-4H/4H] | separate setup & resize from draw(); draw only stacks-view 6c5a1df5 |
| <ul><li>[x] </li></ul> |  | **move to utils/draw/synteny-blocks-draw.js :** 0058f94a |
| <ul><li>[x] </li></ul> | [2H/2H] | showSynteny() ... syntenyIsInverted(), syntenyBlock_2Feature() |
| <ul><li>[x] </li></ul> | | ** change data of axis element from axisID to axis-1d ** e1683595 |
| <ul><li>[x] </li></ul> | [1D] | axis-draw.js |
| <ul><li>[x] </li></ul> | [1D] | stacks.js |
| <ul><li>[x] </li></ul> | [1D] | stacks-drag.js |
| <ul><li>[x] </li></ul> | [1D] | axisBrush.js |
| <ul><li>[x] </li></ul> | [1D] | axisTitle.js |
| <ul><li>[x] </li></ul> | [1D] | axis-menu-actions.js |
| <ul><li>[x] </li></ul> | [1D] | axis-tracks.js |
| <ul><li>[x] </li></ul> | [1D] | block-adj.js, path-data.js |
| <ul><li>[x] </li></ul> | [1-2D] | ... various : axis-ticks-selected.js, axis-menu.js, flows-collate.js, axis.js, drop-target.js, feature-ticks.js, zoomPanCalcs.js |
| <ul><li>[x] </li></ul> |  | **move to axis-1d :** e1683595 |
| <ul><li>[x] </li></ul> | [1H] |  y scale, extracts from ensureAxis(d) e1683595 |
| <ul><li> </li></ul> | [1H/0H] | removeAxis() |
| <ul><li> </li></ul> | [1H/0H] | removeStack() |
| <ul><li> </li></ul> | [2H/0H] | removeAxisMaybeStack() |
| <ul><li>[x] </li></ul> | [2H] | collateO() e1683595 |
| <ul><li>[x] </li></ul> |   [6H] | support selected interface dependencies of functions being moved. e1683595 |
| <ul><li>[x] </li></ul> | [2H] |  change stacks-drag to use stacks-view / axis-1d in place of Stack / Stacked. e1683595 |
| <ul><li>[x] </li></ul> | [8H] | change AxisBrushZoom to use stacks-view / axis-1d in place of Stack / Stacked). e1683595 |
| <ul><li>[x] </li></ul> | [1-2D/8D] | the above changes are working, but expect that more issues will be found as the whole application is able to be tested.  c9d7c9d2 split axis : re-connect .extendedWidth() so that paths have correct left x position when left axis opens.  86afec6f fix id on axis-2d clipPath. de248a76 Don't centre axis ticks text. fdd7603a update paths when axis flipped. fd7872c7  Enable storePath() to lookup blocks for checkFa(), alias paths display OK. 9f5a829d re-connect titleTextMax().  Filter out component after axis is closed. 2023Jan : 300c4f07 : initial value of position in axis-1d. 2029e394 : when tab is not selected, fix 0 features / paths tab count. bede8639 improve wait for definition of yScale. 6c627df0 Re-connect update of axis Y position after dropIn. ccbac32d Don't reduce height of stacked axis twice. c3015201 Ensure highlight of feature circle for path end with numeric feature name. 2052b8f5 re-connect stacks reference used in xDropOutDistance calculation. 60cc0419 stacks-drag : defer change of stacks x position until user completes dragging. 881fa5e5 remove blockIsUnviewed(); re-connect blockVisible() update of selectedFeatures. cfdad5d5 axis brush zoom : don't jump to brush interval when mouse-wheel zoom. 5776b7fd fix a couple of exceptions seen for vcfGenotype. c5bffdc5 Re-enable the VCF Lookup button. 1230a2aa use .set for axis1d.brushedRegion when Reset. 9d1557d6 revert axis position after a drag which did not make a change. 49d3ff42 wait for final axis x position when updating axis menu x position after a drag which changed axis positions.  |
| <ul><li>[x] </li></ul> |  | **move to stacks-view :** 8ccf494d move to stacks-view.js : stacksAdjustY(), stacksAdjust(), axisStackChangedY(), axisStackChanged(). |
| <ul><li>[x] </li></ul> | [1H/1H] | stacksAdjustY |
| <ul><li>[x] </li></ul> | [1H/1H] | stacksAdjust |
| <ul><li>[x] </li></ul> | [/8H] | **move to graph-frame :** f81e5367 moved resize and frame functions out of draw-map as component graph-frame |
| <ul><li> </li></ul> | [1H/0H] | extracts from axisWidthResizeRight() |
| <ul><li>[x] </li></ul> | [1H/1H] | recordViewport() |
| <ul><li>[x] </li></ul> | [2H/2H] | showResize() |
| <ul><li>[x] </li></ul> | [2H/2H] | resizeEffect() |
| <ul><li>[x] </li></ul> | [2H/2H] | stacksWidthChanges() |
| <ul><li>[x] </li></ul> | [2H/2H] | stacksWidthChanged() |
| <ul><li>[x] </li></ul> |  | blockIsUnviewed() (parts, if required) 881fa5e5 remove blockIsUnviewed(); ... |
| <ul><li>[x] </li></ul> |  | **move to added utils/draw/axis-menu :**  removed - replaced by axisTitle : configureAxisTitleMenu()  c238e067 removed functions formerly used by previous configureAxis*TitleMenu() |
| <ul><li>[x] </li></ul> | [1H] | apTitleSel |
| <ul><li>[x] </li></ul> | [1H] | glyphIcon() |
| <ul><li>[x] </li></ul> | [1H] | iconButton() |
| <ul><li>[x] </li></ul> | [2H/2H] | review draw-map for un-required code and remove. b8b02d27 drop code blocks not required after merging stacks classes into corresponding Components |

**Comment by Don-Isdale** [comment]
2022Sep16
---

parallel to the re-organisation, also :
- reduce console.log() trace which is not required.

**Comment by Don-Isdale** [comment]
#### Overview

##### Description
- integrate the stacks & axes more closely with the Ember components which are now the driving force of the app.


##### Background
Stacks & axes were written as javascript classes, using imperative update;  as the number of components and interactions grew, 
a shift was made to implement new components using functional reactive logic (Computed Properties), which is not as efficient as imperative logic, but it scales better with many complex interactions - as long as the connections and dependencies are declared correctly, it will automatically update / recalculate when required, using lazy evaluation.
For the task of moving functions out of draw-map into separate components : most of these functions are related to stacks & axes, so it is best to start by integrating stacks & axes functions into the existing components (stacks-view and axis-1d), so that they are acting merely as utility / library functions, rather than providing imperative actions driving the app event flow.
This change is going well, the core part is done.


##### Methodologies
Moving a function requires re-establishing the input and output data connections, and
providing the facilities which the function being moved depends on.

To some extent it is useful to provide a facsimile of those dependencies / affordances, based on the new design; where there are many uses of a particular dependency, this can save time.
That approach can be transitional;  if changing the interfaces improves the design, that can be done progressively at lower risk.

This method is used when it is important to minimise the implementation effort and the risk of changes requiring secondary changes,
which is not really the case in this instance;  may use this technique for some transitional scaffolding, with the benefit that having the GUI functional is useful for testing.
(for interest, this method was labelled Strangler Fig by Martin Fowler, after he visited Qld rainforests).
(detail of this scaffolding has been prepended to #190 first [comment](https://github.com/plantinformatics/pretzel/issues/190#issue-674013386) )

##### Re-use / Re-write decisions
Each of the functions was evaluated in terms of its value - whether it was exactly or close to the final requirement;  this enables a decision about the degree of reuse / rewrite.
This task #190 lists the functions which are to be moved (i.e. retained), and at the bottom of the list there is a section labelled 'not required' - these functions provide features which are equivalently provided by newer and preferred facilities.
Some other functions have notes indicating how they can be simplified;  the remainder require only the changes for inputs / output and supporting interfaces, which as discussed can be relatively "mechanical".

The functions are listed roughly in parallel to the order in which they appear in the source code, for more efficient work and to check completeness.
As the work is done they can be listed in checked-off order, for easier reference by management to the progress.

##### Risks
. the amount of changes in the functions being moved may be greater than expected
  To quantify this, we can analyse in more detail the supporting dependencies required by functions being moved, list these and a design for supporting them in their new position.
  This has already done with a quick read through of the functions, noting their requirements and thinking about how they would be supported - there were none of any signicant difficulty.


##### Contingency management
###### Fallback plan
If a high priority feature / change is requested, or if time blows out, stash the branch and continue on to other work without merging any changes;
the work is still intact, and can be merged as time permits - it is desirable to minimise the delay in doing this because the merge gets harder as development continues on the feature branches.

#### Planning
There are various approaches to scheduling software maintenance :
. if it is embedded in feature work, it may seem to management feeling that apparently simple features become black holes consuming time
. some tasks can't be done in pieces, and putting it on the schedule as a block of work makes it difficult to get time allocated, as even if it is seen as important, feature development work is more urgent.

**Comment by Don-Isdale** [comment]
This phase of software reorganisation is complete, so closing this item.
Having moved the functions into modules, the following phase #345 will work on those individually and refine the internal and external module interconnection, and replace use of original data structures with use of the model / component / service structures.
